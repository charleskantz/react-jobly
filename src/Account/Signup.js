import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

/**
 *
 * @param {Function} signup: handles new user signup call to DB
 */
function Signup({ signup }) {

  const history = useHistory();
  const [ formData, setFormData ] = useState({
		username: "",
		password: "",
		first_name: "",
		last_name: "",
		email: "",
  });
  const [ formErrors, setFormErrors ] = useState([]);

  const handleSignup = async evt => {
    evt.preventDefault();
    let result = await signup(formData);
    if (result.login) {
      history.push("/companies");
    } else {
      setFormErrors(result.err);
    }
  };

  const handleChange = evt => {
		const { name, value } = evt.target;
		setFormData(data => ({ ...data, [name]: value }));
	};

  return (
    <div>
      <form className="SignUpForm" onSubmit={handleSignup}>
        <div>
          <label htmlFor="username">Username: </label>
          <input
            onChange={handleChange}
            name="username"
            placeholder="username"
            value={formData.username}
            id="username"
          />
        </div>
        <div>
          <label htmlFor="password">Password: </label>
          <input
            onChange={handleChange}
            type="password"
            name="password"
            placeholder="password"
            value={formData.password}
            id="password"
          />
        </div>
        <div>
          <label htmlFor="first_name">First Name: </label>
          <input
            onChange={handleChange}
            name="first_name"
            placeholder="firstName"
            value={formData.first_name}
            id="firstName"
          />
        </div>
        <div>
          <label htmlFor="last_name">Last Name: </label>
          <input
            onChange={handleChange}
            name="last_name"
            placeholder="lastName"
            value={formData.last_name}
            id="lastName"
          />
        </div>
        <div>
          <label htmlFor="email">Email: </label>
          <input
            onChange={handleChange}
            name="email"
            placeholder="email"
            value={formData.email}
            id="email"
          />
        </div>

        {formErrors.length
          ? <p>{formErrors}</p>
          : null
        }

        <button id="addNewUser">Create Account</button>
      </form>
    </div>
  )
}

export default Signup;