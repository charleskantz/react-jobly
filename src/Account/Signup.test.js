import React from "react";
import { render } from "@testing-library/react";
import Signup from "./Signup";

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
  Link: 'a'
}));

// smoke
it("renders without crashing", function() {
  render(
    <Signup />
  );
});

// snapshot
it("matches snapshot", function() {
  const { asFragment } = render(<Signup />);
  expect(asFragment()).toMatchSnapshot();
});