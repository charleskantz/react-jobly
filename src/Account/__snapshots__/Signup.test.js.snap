// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-omwz41"
  >
    <h1
      class="css-1awjs8x"
    >
      Sign Up
    </h1>
    <p
      class="css-1s86eya"
    >
      All fields are required.
    </p>
    <div
      class="css-orpbuy"
    >
      <form>
        <label
          class="css-182h8ta"
          for="username"
        >
          Username
        </label>
        <input
          autocomplete="username"
          class="css-1dltt36"
          name="username"
          value=""
        />
        <label
          class="css-182h8ta"
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="new-password"
          class="css-1dltt36"
          name="password"
          type="password"
          value=""
        />
        <label
          class="css-182h8ta"
          for="email"
        >
          Email
        </label>
        <input
          autocomplete="email"
          class="css-1dltt36"
          name="email"
          type="email"
          value=""
        />
        <label
          class="css-182h8ta"
          for="first_name"
        >
          First Name
        </label>
        <input
          autocomplete="given-name"
          class="css-1dltt36"
          name="first_name"
          value=""
        />
        <label
          class="css-182h8ta"
          for="last_name"
        >
          Last Name
        </label>
        <input
          autocomplete="family-name"
          class="css-1dltt36"
          name="last_name"
          value=""
        />
        <div
          class="css-303viu"
          display="flex"
        >
          <button
            class="css-1xrqucq"
            type="submit"
          >
            Sign Up
          </button>
          <div
            class="css-1fkcxxx"
            display="flex"
          >
            <p
              class="css-1s86eya"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
